### DISCUSSION TITLE
Improvements in Contribution Guide

### DISCUSSION THREAD

**Discussion by estuardodev** [reactive-python/reactpy-django#161] [Ideas]
I feel that there should be some rule-like form of what the branches that will be created when making some modification should be called. For example:

The example.py file is modified by a misspelled comment

And an issues called `"Wrong comment on example.py" #001` is created 

To create the branch I feel it should be type:
`git branch bug-example`

and to send the commit it should be something like:
` git commit -m "Bug example #001"`

So have more clear and orderly that is done.

In addition to everything indicate when a CHANGELOG should be made and what steps should be followed.

**Comment by Archmonger** [comment]
I don't think it should become our responsibility to explain to the end user how to use the `git` CLI.

In general, we assume that GitHub contributors have learned these skills from elsewhere. I don't mind adding a `tip` that contains a link to some tutorials about Git.

But, it's probably easier to explain to beginners how to use GitHub desktop rather than the GitHub CLI though.

**Comment by estuardodev** [comment]
I don't mean to explain to the user how to use Git. Rather to indicate some way to follow how to create the branches or make the messages of the commits.

Example:
When creating the branches they should carry if it is a mistake, it is a bug or it is an idea. Followed by the name of the file to be modified.
Names:
```
bug-example
error-index
idea-example
```

And the messages of the commits should carry the number of the issues type:
`"Fixed bug #001"`

**Reply by Archmonger** [reply]
Since we squash merge all PRs, we don't need to structure how the user formats their commits. We generally prefer fixed issues to be referenced in the PR description.

The name of the forked branch doesn't exactly matter either.
